File: nomad-samples/js/account-registration/account/verify.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function verify(EMAIL, TOKEN) 
{
		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
  
  	const BODY = {
        userName: EMAIL,
        token: TOKEN
    };
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/verify`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        console.log(exception);
        throw exception;
    });
    
    // Check for success
    if (RESPONSE.ok) {
        console.log("Success");
        return(RESPONSE.status);
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, `Verify user failed`);
}

File: nomad-samples/common/py/account_registration/account/verify.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

def verify(EMAIL: str, CODE: int) -> dict:
    if not EMAIL:
        raise Exception("Email: The email is invalid")
    
    if not CODE:
        raise Exception("Code: The code is invalid")
    
    API_URL = PORTAL_API_URL + "/account/verify"

    HEADERS = {
      "Content-Type": "application/json"
    }
    
    # replace username and password with your username and password
    BODY = {
    	"userName": EMAIL, 
        "token": CODE
    }

    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)
    except:
      api_exception_handler(RESPONSE, "Verification failed")
            
